ui/app/models/task.js

```diff
@@ -79,4 +82,20 @@ export default class Task extends Fragment {
       );
     }
   }
+
+  // TODO: This async fetcher seems like a better fit for most of our use-cases than the above getter (which cannot do async/await)
+  async getPathLinkedVariable() {
+    if (!this._job) {
+      await this._fetchParentJob();
+    }
+    await this._job.variables;
+    let jobID = this._job.plainId;
+    if (this._job.parent.get('plainId')) {
+      jobID = this._job.parent.get('plainId');
+    }
```

Comment:
Is this safe? Looking at this, it seems like this would be big mad if `_job.parent` wasn't defined (or maybe it's always defined because it's a relationship?) and it would also potentially erroneously return `undefined` because it has to load the parent job first, in the case there is a parent job that isn't resolved yet.

You can at least get the parent job's ID synchronously and without making a request with

```js
this._job.belongsTo('parent').id()
```

But I believe this would take the `['job-id','namespace']` form.

Reply:
Hm yeah good catch, I guess parentlessJob.parent.get('x') doesn't error out by virtue of ember relationships. I'm going to switch this up regardless, so thanks for the suggestion.

Reply:
Swapped to this:

```
  async getPathLinkedVariable() {
    if (!this._job) {
      await this._fetchParentJob();
    }
    await this._job.variables;
    let jobID = this._job.plainId;
    // not getting plainID because we dont know the resolution status of the task's job's parent yet
    let parentID = this._job.belongsTo('parent').id()
      ? JSON.parse(this._job.belongsTo('parent').id())[0]
      : null;
    if (parentID) {
      jobID = parentID;
    }
    return await this._job.variables?.findBy(
      'path',
      `nomad/jobs/${jobID}/${this.taskGroup.name}/${this.name}`
    );
  }
```